eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -304,30 +303,26 @@ object MultiPartPaymentLifecycle {
    * Send a payment to a given node. The payment may be split into multiple child payments, for which a path-finding
    * algorithm will run to find suitable payment routes.
    *
-   * @param paymentSecret   payment secret to protect against probing (usually from a Bolt 11 invoice).
-   * @param targetNodeId    target node (may be the final recipient when using source-routing, or the first trampoline
-   *                        node when using trampoline).
-   * @param totalAmount     total amount to send to the target node.
-   * @param targetExpiry    expiry at the target node (CLTV for the target node's received HTLCs).
-   * @param maxAttempts     maximum number of retries.
-   * @param paymentMetadata payment metadata (usually from the Bolt 11 invoice).
-   * @param extraEdges      routing hints (usually from a Bolt 11 invoice).
-   * @param routeParams     parameters to fine-tune the routing algorithm.
-   * @param additionalTlvs  when provided, additional tlvs that will be added to the onion sent to the target node.
-   * @param userCustomTlvs  when provided, additional user-defined custom tlvs that will be added to the onion sent to the target node.
+   * @param recipient   final recipient.
+   * @param maxAttempts maximum number of retries.
+   * @param routeParams parameters to fine-tune the routing algorithm.
    */
-  case class SendMultiPartPayment(replyTo: ActorRef,
-                                  paymentSecret: ByteVector32,
-                                  targetNodeId: PublicKey,
-                                  totalAmount: MilliSatoshi,
-                                  targetExpiry: CltvExpiry,
-                                  maxAttempts: Int,
-                                  paymentMetadata: Option[ByteVector],
-                                  extraEdges: Seq[ExtraEdge] = Nil,
-                                  routeParams: RouteParams,
-                                  additionalTlvs: Seq[OnionPaymentPayloadTlv] = Nil,
-                                  userCustomTlvs: Seq[GenericTlv] = Nil) {
-    require(totalAmount > 0.msat, s"total amount must be > 0")
+  case class SendMultiPartPayment(replyTo: ActorRef, recipient: Recipient, maxAttempts: Int, routeParams: RouteParams) {
+    require(recipient.totalAmount > 0.msat, "total amount must be > 0")
+
+    val targetNodeId: PublicKey = recipient match {
+      // When using trampoline, we only need to find a route to the first trampoline node, not the final node.
+      case r: TrampolineRecipient => r.trampolineNodeId
+      case r => r.nodeId
+    }
+    val amountToSend: MilliSatoshi = recipient match {
+      case r: TrampolineRecipient => r.trampolineAmount
+      case r => r.totalAmount
+    }
```

Comment:
This should be hidden inside the `TrampolineRecipient`. Are `r.nodeId` and `r.totalAmount` ever used if `r` is a `TrampolineRecipient`?

Reply:
Same as above, I kept this currently to minimize changes because refactoring trampoline payments deserves its own PR. I fully agree with you, and it will be done!

Reply:
Done.